src/WebJobs.Script/Diagnostics/DefaultLoggerProviderFactory.cs

```diff
@@ -30,14 +30,17 @@ public virtual void AddLoggerProviders(ILoggerFactory factory, ScriptHostConfigu
                 ITelemetryClientFactory clientFactory = scriptConfig.HostConfig.GetService<ITelemetryClientFactory>() ??
                     new ScriptTelemetryClientFactory(settingsManager.ApplicationInsightsInstrumentationKey, scriptConfig.LogFilter.Filter);
 
-                scriptConfig.HostConfig.LoggerFactory.AddApplicationInsights(clientFactory);
+                providers.Add(new ApplicationInsightsLoggerProvider(clientFactory));
             }
             else
             {
                 metricsLogger?.LogEvent(MetricEventNames.ApplicationInsightsDisabled);
             }
 
-            factory.AddConsole(scriptConfig.LogFilter.DefaultLevel, true);
+            providers.Add(new FileLoggerProvider(scriptConfig.RootLogPath, isFileLoggingEnabled, isPrimary));
```

Comment:
One of these is the user file, and one is the host file logger? If so, maybe the non host one should be called FunctionFileLoggerProvider?

Reply:
Yes, it should. That'd be clearer.